src/Grpc.Net.Client/Internal/GrpcProtocolConstants.cs

```diff
@@ -103,15 +115,29 @@ static GrpcProtocolConstants()
             // but in case it isn't then don't include version in user-agent.
             if (assemblyVersion != null)
             {
-                userAgent += "/" + assemblyVersion.Version;
+                // 1.24.0
+                userAgent += $"/{assemblyVersion.Version} ";
             }
 
-            UserAgentHeader = "User-Agent";
-            UserAgentHeaderValue = userAgent;
-            TEHeader = "TE";
-            TEHeaderValue = "trailers";
+            // (.NET 5.0.7;
+            userAgent += $"({RuntimeInformation.FrameworkDescription}; ";
+            // CLR 5.0.0;
+            userAgent += $"CLR {Environment.Version}; ";
+
+            // .NETCoreApp,Version=v6.0; 
+            var targetFramework = assembly
+                .GetCustomAttributes<TargetFrameworkAttribute>()
+                .FirstOrDefault()?
+                .FrameworkName;
+            if (!string.IsNullOrEmpty(targetFramework))
+            {
+                userAgent += $"{targetFramework}; ";
+            }
 
-            DefaultMessageAcceptEncodingValue = GetMessageAcceptEncoding(DefaultCompressionProviders);
+            // x64)
+            userAgent += Environment.Is64BitProcess ? "x64)" : "x32)";
+
+            return userAgent;
```

Comment:
we could really use info about the operating system / platform. 
Grpc.Core provides that info in the second part of the user agent string that corresponds to C-core (`grpc-csharp/2.38.0-dev (.NET 5.0.0; CLR 5.0.0; netstandard2.0; x64) grpc-c/16.0.0 (osx; chttp2)`) - but since grpc-dotnet doesn't have the C-core layer, we need to include platform info ourselves.  Use `RuntimeInformation.IsOSPlatform()`?